format/src/main/scala/org/scalafmt/internal/SyntaxTokens.scala

```diff
@@ -282,6 +268,112 @@ object SyntaxTokens {
     def tokensWith: KwWith = tree.findBetween[KwWith](_.lhs, _.rhs).get
   }
 
+  // == Defn ==
+  implicit class XtensionDefnClassSyntax(val tree: Defn.Class) extends AnyVal {
+    def tokensClass: KwClass = tree.find[KwClass].get
+
+    // tparams
+    def tokensLeftBracket: Option[LeftBracket] =
+      tree.tparams.headOption
+        .map(tparam => tree.findBetween[LeftBracket](_.name, _ => tparam).get)
+    def tokensRightBracket: Option[RightBracket] =
+      tree.tparams.lastOption.map(
+        tparam =>
+          tree.ctor.paramss match {
+            case Nil => tree.findAfter[RightBracket](_ => tparam).get
+            case xs :: _ =>
+              xs match {
+                case Nil => tree.findAfter[RightBracket](_ => tparam).get
+                case x :: _ =>
+                  tree.findBetween[RightBracket](_ => tparam, _ => x).get
+              }
+        }
+      )
+
+    def tokensCommaTparams: List[Comma] =
+      commaSeparated(tree)(_.tparams)
+
+    def tokensCommaCtor: List[List[Comma]] =
```

Comment:
This should be an extension method on `Ctor.Primary` instead of `Defn.Def`

Reply:
We can do this later, currently, it's coupled with the starting token.